dialogue-client-test-lib/src/main/java/com/palantir/dialogue/AbstractChannelTest.java

```diff
@@ -196,165 +192,122 @@ public void fillsQueryParameters() throws Exception {
         assertThat(server.takeRequest().getMethod()).isEqualTo("POST");
     }
 
     @Test
     public void putMethodYieldsPutHttpCall() throws InterruptedException {
         endpoint.method = HttpMethod.PUT;
         when(request.body()).thenReturn(Optional.of(body));
-        channel.createCall(endpoint, request).execute(observer);
+        channel.createCall(endpoint, request);
         assertThat(server.takeRequest().getMethod()).isEqualTo("PUT");
     }
 
     @Test
     public void deleteMethodYieldsDeleteHttpCall() throws InterruptedException {
         endpoint.method = HttpMethod.DELETE;
-        channel.createCall(endpoint, request).execute(observer);
+        channel.createCall(endpoint, request);
         assertThat(server.takeRequest().getMethod()).isEqualTo("DELETE");
     }
 
     @Test
     public void callObservesSuccessfulResponses() {
-        channel.createCall(endpoint, request).execute(observer);
-        verify(observer).success(any());
-        verify(observer, never()).failure(any());
-        verify(observer, never()).exception(any());
-    }
-
-    @Test
-    public void cannotExecuteCallTwice() {
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        assertThatThrownBy(() -> call.execute(observer))
-                .hasMessage("Error, this call was already executed");
-    }
-
-    @Test
-    public void canCancelNonExecutedCalls() {
-        assertThatCode(() -> channel.createCall(endpoint, request).cancel()).doesNotThrowAnyException();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        Futures.addCallback(call, callback, MoreExecutors.directExecutor());
+        verify(callback).onSuccess(any());
+        verify(callback, never()).onFailure(any());
     }
 
     @Test
     public void canCancelMultipleTimes() {
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        assertThatCode(call::cancel).doesNotThrowAnyException();
-        assertThatCode(call::cancel).doesNotThrowAnyException();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        assertThatCode(() -> call.cancel(true)).doesNotThrowAnyException();
+        assertThatCode(() -> call.cancel(true)).doesNotThrowAnyException();
     }
 
     @Test
     public void callCancellationIsObservedAsException() throws InterruptedException {
-        channel.createCall(endpoint, request).execute(mock(Observer.class));  // drain enqueued response
+        channel.createCall(endpoint, request);  // drain enqueued response
 
         channel = createChannel(server.url("").url(), Executors.newCachedThreadPool());
-        Call call = channel.createCall(endpoint, request);
-        call.execute(observer);
-        call.cancel();
+        ListenableFuture<Response> call = channel.createCall(endpoint, request);
+        Futures.addCallback(call, callback, MoreExecutors.directExecutor());
```

Comment:
as above, this doesn't make much sense now

Reply:
done